src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -392,6 +392,17 @@ internal async void SetSignInStatus(bool status)
                 $"signInStatus: \"" + status + "\"})");
             }
         }
+        /// <summary>
+        /// Enable or not the SignIn button on the fly.
+        /// </summary>
+        /// <param name="enabled"></param>
+        internal async void SetSignInEnable(bool enabled)
```

Comment:
You may need to expose this as an API and provide an example of how to use it as part of this PR

Reply:
Sure, currently is just a method of the SplashScreen class, Do you want to extend it as part of the DynamoViewModel, like the RequestClose ?

Reply:
> You may need to expose this as an API and provide an example of how to use it as part of this PR

Sure, currently is just a method of the SplashScreen class, Do you want to extend it as part of the DynamoViewModel, like the RequestClose ?

Reply:
Please think about it in the integration context: https://github.com/DynamoDS/DynamoRevit/blob/master/src/DynamoRevit/DynamoRevit.cs#L334 how to disable it when trying to show it from integrations.

Reply:
> You may need to expose this as an API and provide an example of how to use it as part of this PR

@QilongTang @mjkkirschner @reddyashish , changing it to public is enough to expose it so can be called from other parts. Unfortunatelly after varioous tests on D4R deptly I have realized its method OnRequestDynamicSplashScreen is called before the splash creen is ready and it's being used by Revit on that way which IMHO is a bug because the splash screen is ready only when its web resources are completed loaded and ready to have access to the WV2's DOM. That's why an example over the current implementation of the flow between Dynamo and D4R doesn't work. Please let me know What Do you consider to complete this PR, thanks.

![image](https://github.com/DynamoDS/Dynamo/assets/96534278/26ae3f81-4823-4ab4-89da-8d781039ebf3)

Reply:
Why are we sticking to a method, not some startup config for SplashScreen constructor? A method call is meant to fail after UI initialized?

Reply:
> Why are we sticking to a method, not some startup config for SplashScreen constructor? A method call is meant to fail after UI initialized?

A method approach means the consumers can call it on demand / Yes because the consumers don't know when the UI is initialized properly (image above).

I have moved the responsibility to enable or not the SignIn button to the ctor so the consumers will be able to decide only at that time.

Now the Splashcreen is the responsible to enable or not the SignIn button only when the web resources are completely loaded.

The ctor will enable the SignIn button by default, otherwise it needs to set explicitly here https://github.com/DynamoDS/DynamoRevit/blob/master/src/DynamoRevit/DynamoRevit.cs#L334 according to the consumer's criteria.

![image](https://github.com/DynamoDS/Dynamo/assets/96534278/89986ee2-b2c8-424a-9399-b0f76ba0a8b1)

Enabled
![login1](https://github.com/DynamoDS/Dynamo/assets/96534278/14518231-2128-47fa-b9b5-0acf1f112bb6)

Disabled
![login2](https://github.com/DynamoDS/Dynamo/assets/96534278/32bf1e1e-f5c1-46b3-9dfe-23a2f1941a95)